gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskConfiguration.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.api.services.compute.model.Disk;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for Google Compute Engine disk configurations. A disk can be used as primary storage
+ * for your virtual machine instances. Use {@link StandardDiskConfiguration} to create a standard
+ * disk given a disk type and size. Use {@link ImageDiskConfiguration} to create a disk from a
+ * Compute Engine disk image. Use {@link SnapshotDiskConfiguration} to create a disk from a Compute
+ * Engine disk snapshot.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/">Block Storage</a>
+ */
+public abstract class DiskConfiguration implements Serializable {
+
+  private static final long serialVersionUID = -1783061701255428417L;
+
+  private final Type type;
+  private final Long sizeGb;
+  private final DiskTypeId diskType;
+
+  /**
+   * Type of a Google Compute Engine disk configuration.
+   */
+  public enum Type {
+    /**
+     * A Google Compute Engine standard disk configuration.
+     */
+    STANDARD,
+
+    /**
+     * A Google Compute Engine disk configuration that creates a disk from an image.
+     */
+    IMAGE,
+
+    /**
+     * A Google Compute Engine disk configuration that creates a disk from a snapshot.
+     */
+    SNAPSHOT
+  }
+
+  /**
+   * Base builder for disk configurations.
+   *
+   * @param <T> the disk configuration type
+   * @param <B> the disk configuration builder
+   */
+  public abstract static class Builder<T extends DiskConfiguration, B extends Builder<T, B>> {
+
+    private Type type;
+    private Long sizeGb;
+    private DiskTypeId diskType;
```

Comment:
I believe this is consistent with what we did so far. Quoting @ajkannan from our previous discussion on this

> I see, I think @mziccard uses [class]Id() and id() in [class].java, but anywhere there's a field that points to the identity of another class, he uses [class] instead of [class]Id, probably for the sake of brevity. In this case, @mziccard's code as it is is consistent with other usage.